app/controllers/concerns/bulk_upload_handler.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module BulkUploadHandler
+  def handle_bulk_upload(options)
```

Comment:
Is there a reason for using a hash instead regular parameters? Can we just use regular parameters for this. e.g.
```
def handle_bulk_upload(:file, :error_class, :validator_class)
  ...
end
```

Reply:
Using a hash would be cleaner and maintainable when calling `handle_bulk_upload` from different controllers,  and makes it easier to extend the method without breaking existing calls.